src/OrchardCore.Cms.Web/OrchardCore.Cms.Web.csproj

```diff
@@ -6,7 +6,6 @@
   <Import Project="..\OrchardCore\OrchardCore.Application.Cms.Core.Targets\OrchardCore.Application.Cms.Core.Targets.props" />
 
   <PropertyGroup>
-    <TargetFramework>$(AspNetCoreTargetFramework)</TargetFramework>
```

Comment:
Have you tried with : 

`<TargetFrameworks>$(AspNetCoreTargetFrameworks)</TargetFrameworks>`

Reply:
@Skrypt On line 3, the project imports `..\OrchardCore.Build\Dependencies.props`, which in turn imports `Dependencies.AspNetCore.props`.  Due to that, the `<TargetFrameworks>` property is already set which makes this redundant.

Reply:
Ok, though if I want to compile this assembly/module only "outside of OC build context" ; I then need to keep it in that folder so that it gets compiled. Ok, I don't mind. I'm pretty sure there was a reason that we kept it redundant but we will see. Keep it removed for now, no big issue there.

Reply:
Yes, I thought of that.  That's the reason that I split out the commits for removing it by first commenting it in the initial commit that made the TFM change, and then in a second commit removing the comments (and some inconsquential whitespace that my editor made due to the standard setting of removing trailing whitespace).  That way we can revert the second commit easily and then do a find and replace without removing the changes in the `Dependencies.AspNetCore.props` file.

I think a lot of the projects will not build properly outside of the OC build context due to the `<PackageManagement>` tasks set up to control the package versions and common dependencies at a central location.  It took me a bit to wade through the build dependency chain and understand what was going on.

Reply:
@dafergu2 

@Skrypt is right, and our web project file is also an example of the needed properties of a web project file

Personally i would have kept `$(AspNetCoreTargetFramework)`, but replaced by `$(AspNetCoreTargetFrameworks)`, in all project files, and kept the definition of `AspNetCoreTargetFramework` and add `AspNetCoreTargetFrameworks` in the global `Dependencies.AspNetCore.props`, so that each individual project can choose

I already tried multi targetting with OrchardCore, but a while ago so, for the `PackagePath` of msbuild scripts, i don't remember if i kept the `$(TargetFramework)` as all project are inner built 2 times but with the right TFM each time, or maybe if i used a `buildMultiTargetting` folder as aspnetcore does.

Need to focus on it again, then i will review it

Reply:
@jtkech I had originally done that, but as I as was understanding the flow of the build tasks, it seemed to add more to confusion to have those extra prop settings.  An easy to miss typo with that also caused a significant amount of time to be consumed tracking it down when I was testing.  Each project can still override the `<TargetFramework>` property, it would just need to be added as `<TargetFramework>net5.0</TargetFramework>` or similar.

When multi-targeting, the `OrchardCore.Cms.Web` project will be built for both.  If you try to run it, it indicates that the framework needs to be specified with --framework.  Should there be a default setting or should the user specify?  I haven't tried building and running on an environment that has only the 3.1 framework or 5.0 framework installed, so I'm unsure of the exact behavior at that point.

Anecdotally, confusion on the $(AspNetCoreTargetFramework) MSBuild var was expressed by a coworker on my current project which we are implementing using Orchard Core.

I'm happy to add back in the properties if that's desired, though.  I wanted to walk through some of my thought process to show the change was made after some deliberation and not just at a whim. :)  All is good, just let me know and I'll make any changes desired.

With regards to multi-targeting and packaging, this was the larger challenge as it involved figuring out how the `Directory.Build.<targets|props>` files were working for the Modules and Themes packages.  The `PackagePath` was causing the assembly props file to be copied from the `src\OrchardCore\OrchardCore.Module.Targets\Package.Build.props`  to `build\netcoreapp3.1\<Assembly>.props` and `build\net5.0\<Assembly>.props` respectively for each TFM.  This was causing the nuget validation to issue a NU5129 error in which it complained that the `build\` folder (without a TFM subsegment) had a .props file but not one named in the convention of the <Assembly>.props naming convention.  By removing the TFM segment in the <PackagePath> property, the validation was fixed.  This makes sense because the nuget package is now packaging together targets for multiple frameworks in a single package rather than separate packages for each TFM.

I didn't find a good explanation of the `build/` folder and the other handful of related folders, though, so if there is a resource that explains which to use when, please let me know.  By moving the .props file to the higher level, it seems to have fixed the problem and the resulting packages seem correct on the surface.

Thanks so much for your quick feedback!  :)

Reply:
The Cms.Web project should have a single target. In that case net5.0. Same for the templates. Then if a user wants to target netcoreapp3.1 they can create their own project.

Reply:
@sebastienros Okay, I'll make that change.  Same thing for the `OrchardCore.Mvc.Web.csproj` and the other web app or console apps in the tests folder.

Reply:
The Cms.Web and the Mvc.Web projects will now default to the `$(AspNetCoreTargetFramework)` prop that is defined in the `OrchardCore.Build\Dependencies.AspNetCore.props` file.  That prop is not set based on the multi-target passes made by MSBuild, but is defined as a string constant.

You may want to look at commit msg for 363324f07e5b8b13f1d672c4decb46015aab32e9 for a little detail about why the Cms.Web is conditionally setting the `<TargetFramework>` prop.  This is specifically because the `OrchardCore.Tests.csproj` references the Cms.Web project.  We want the `OrchardCore.Tests.csproj` to be fully multi-targeted since that will make `dotnet test` run the test suite for all of the targets in `<TargetFrameworks>` instead of having to manually do it for each target.  If you look at the change in the `OrchardCore.Tests.csproj`, you'll notice that the project reference item is specifying a `SetTargetFramework` metadata attribute.  This will force the correct TFM to be used when referencing from the `OrchardCore.Tests.csproj` file on both the netcoreapp3.1 and net5.0 targets, but allows the Cms.Web project to set it when it is empty and not referenced by the `OrchardCore.Tests.csproj`.

Reply:
The reference by the `OrchardCore.Tests.csproj` is causing the `netcoreapp3.1` TFM target to be built for `Cms.Web` to be built, not the pattern described above, unfortunately.

The functional tests are failing due to them expecting the `netcoreapp3.1` binary of `OrchardCore.Mvc.Web` to be present.  I've updated the cypress et. al js scripts to  use the `net5.0`.  If both are needed, we can reference the `OrchardCore.Mvc.Web.csproj` in the functional test project and specify the target framework in the reference to force it to build.

Reply:
So to follow up again, after more testing (and with lots of cleaning out of obj/bin/temp folders between tests) and solution rebuilds, I believe the defaulting of Cms.Web and Mvc.Web doesn't need to the conditional, so I reworked those settings to be a normal setting.